[Feature][connector-kudu] support Kerberos authentication

NickCodeJourney opened the pull request:
<!--

Thank you for contributing to SeaTunnel! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

## Contribution Checklist

  - Make sure that the pull request corresponds to a [GITHUB issue](https://github.com/apache/seatunnel/issues).

  - Name the pull request in the form "[Feature] [component] Title of the pull request", where *Feature* can be replaced by `Hotfix`, `Bug`, etc.

  - Minor fixes should be named following this pattern: `[hotfix] [docs] Fix typo in README.md doc`.

-->

## Purpose of this pull request

<!-- Describe the purpose of this pull request. For example: This pull request adds checkstyle plugin.-->

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [x] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [x] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

Commit:
:sparkles: 支持kerberos认证

FlechazoW commented on seatunnel-connectors-v2/connector-kudu/pom.xml:
```diff
@@ -56,5 +56,17 @@
+        <dependency>
+            <groupId>org.apache.seatunnel</groupId>
+            <artifactId>connector-file-base</artifactId>
+            <version>2.3.2-SNAPSHOT</version>
```
use ${project.version} .

FlechazoW commented on seatunnel-connectors-v2/connector-kudu/pom.xml:
```diff
@@ -56,5 +56,17 @@
+            <groupId>org.apache.hadoop</groupId>
+            <artifactId>hadoop-client-api</artifactId>
+            <version>3.3.2</version>
+            <scope>compile</scope>
```
What's the reason you add this dependency?

Commit:
:sparkles: Great job! This commit message adequately describes what changes were made and why. It provides context to the change and expresses the reason behind deleting KuduClientUtils and refactoring Kudu Source/Sink Config without being too verbose or complex.

Commit:
:fire: Remove unnecessary CONF_FILES from KuduSourceConfig

This commit removes the unnecessary CONF_FILES option in the KuduSourceConfig. The option was redundant and not being used in the current implementation of the Kudu Connector, thus simplifying the overall configuration.

Commit:
:sparkles: Added Apache License header in KuduClientUtil

Inserted the Apache License 2.0 header at the start of KuduClientUtil.java file. This's necessary to comply with the project's licensing requirements and to clearly state the terms under which the code is distributed.

Commit:
Merge remote-tracking branch 'origin1/dev' into dev-kudukerberos

NickCodeJourney committed:
Merge branch 'apache:dev' into dev-kudukerberros

NickCodeJourney committed:
Merge branch 'apache:dev' into dev-kudukerberros

FlechazoW commented on seatunnel-connectors-v2/connector-kudu/src/main/java/org/apache/seatunnel/connectors/seatunnel/kudu/config/KuduSinkConfig.java:
```diff
@@ -51,13 +51,36 @@ public class KuduSinkConfig {
+    public static final Option<String> CONF_FILES =
+            Options.key("conf.files").stringType().noDefaultValue().withDescription("conf files ");
+
+    public static final Option<String> KERBEROS_PRINCIPAL =
+            Options.key("kerberos_principal")
+                    .stringType()
+                    .noDefaultValue()
+                    .withDescription("Kudu kerberos principal");
+
+    public static final Option<String> KERBEROS_KEYTAB_PATH =
+            Options.key("kerberos_keytab_path")
+                    .stringType()
+                    .noDefaultValue()
+                    .withDescription("Kudu kerberos keytab file path");
```
You can abstract these configuration items into the common module.

FlechazoW commented on seatunnel-connectors-v2/connector-kudu/src/main/java/org/apache/seatunnel/connectors/seatunnel/kudu/config/KuduSourceConfig.java:
```diff
@@ -41,4 +41,22 @@ public class KuduSourceConfig implements Serializable {
                     .noDefaultValue()
                     .withDescription("Specifies the column names of the table");
+
+    public static final Option<Boolean> USE_KERBEROS =
+            Options.key("use.kerberos")
+                    .booleanType()
+                    .defaultValue(false)
+                    .withDescription("kudu use.kerberos");
+
+    public static final Option<String> KERBEROS_PRINCIPAL =
+            Options.key("kerberos_principal")
+                    .stringType()
+                    .noDefaultValue()
+                    .withDescription("Kudu kerberos principal");
+
+    public static final Option<String> KERBEROS_KEYTAB_PATH =
+            Options.key("kerberos_keytab_path")
+                    .stringType()
+                    .noDefaultValue()
+                    .withDescription("Kudu kerberos keytab file path");
```
Same as abrove.

hailin0 reviewed (commented):
Please update documentation and test cases

Carl-Zhou-CN commented:
Now that Kudu has added support for Kerberos, if there are no new updates, this PR will be closed.

NickCodeJourney commented:
agree

NickCodeJourney commented:
> Now that Kudu has added support for Kerberos, if there are no new updates, this PR will be closed.

agree
